core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -524,18 +535,70 @@ public boolean hasLicensesXml() {
      */
     /*package*/ void resolvePluginDependencies() throws IOException {
         List<String> missingDependencies = new ArrayList<String>();
+        List<String> obsoleteDependencies = new ArrayList<String>();
+        List<String> disabledDependencies = new ArrayList<String>();
         // make sure dependencies exist
         for (Dependency d : dependencies) {
-            if (parent.getPlugin(d.shortName) == null)
+            PluginWrapper dependency = parent.getPlugin(d.shortName);
+            if (dependency == null) {
                 missingDependencies.add(d.toString());
+            } else {
+                if (!dependency.isActive()) {
+                    disabledDependencies.add(d.toString());
+                }
+                if (ENABLE_PLUGIN_DEPENDENCIES_VERSION_CHECK) {
+                    if (dependency.getVersionNumber().isOlderThan(new VersionNumber(d.version))) {
+                        obsoleteDependencies.add(dependency.getShortName() + "(" + dependency.getVersion() + " < " + d.version + ")");
+                    }
+                }
+            }
         }
-        if (!missingDependencies.isEmpty())
-            throw new IOException("Dependency "+Util.join(missingDependencies, ", ")+" doesn't exist");
-
         // add the optional dependencies that exists
         for (Dependency d : optionalDependencies) {
-            if (parent.getPlugin(d.shortName) != null)
-                dependencies.add(d);
+            PluginWrapper dependency = parent.getPlugin(d.shortName);
+            if (dependency != null) {
+                boolean valid = true;
+                if (!dependency.isActive()) {
+                    LOGGER.warning(d + " is disabled. This plugin won't be able to access its classes.");
```

Comment:
With proper log context, I think it is okay, but of course this is only my opinion.
```
Mar 29, 2016 7:43:13 PM hudson.PluginManager dynamicLoad
INFO: Attempting to dynamic load /Users/vincent/Projects/Opensource/jenkins/test/target/hudson3353537990711553137test/plugins/depender-0.0.2.hpi
Mar 29, 2016 7:43:13 PM hudson.PluginWrapper resolvePluginDependencies
WARNING: dependee (0.0.2) optional is disabled. This plugin won't be able to access its classes.
```